pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StAreaFunction.java

```diff
@@ -62,8 +63,8 @@ public void init(List<TransformFunction> arguments, Map<String, DataSource> data
     TransformFunction transformFunction = arguments.get(0);
     Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
         "Argument must be single-valued for transform function: %s", getName());
-    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES,
-        "The argument must be of bytes type");
+    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES
+        || transformFunction instanceof LiteralTransformFunction, notBytesErrorMessage(transformFunction));
```

Comment:
i am not sure this logic is correct. this means if transformFunction is Literal, you don't check for the DataType of the result?

Reply:
@walterddr yeh this is a good question! But all the existing ST_* functions do it like this and it seems to work alright.

Reply:
@walterddr see an example here of what I mean - https://github.com/apache/pinot/blob/master/pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StContainsFunction.java#L60

Reply:
ahh. that's the obvious reason: https://github.com/apache/pinot/blob/ca089681483bb3066890d0ff9b4cea25428b1d12/pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java#L39